azurerm/internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -177,14 +205,20 @@ func resourceArmKustoClusterCreateUpdate(d *schema.ResourceData, meta interface{
 
 	location := azure.NormalizeLocation(d.Get("location").(string))
 
-	sku, err := expandKustoClusterSku(d)
+	sku, err := expandKustoClusterSku(d.Get("sku").([]interface{}))
 	if err != nil {
 		return err
 	}
 
 	zones := azure.ExpandZones(d.Get("zones").([]interface{}))
 
+	optimizedAutoScale := expandOptimizedAutoScale(d.Get("optimized_auto_scale").([]interface{}))
+	if optimizedAutoScale != nil {
+		sku.Capacity = optimizedAutoScale.Minimum
```

Comment:
We set the value of capacity to the value of `minimum_instances`. Capacity is a required attribute, although it will be set to the value of `minimum_instances` after creating or updating the resource.